test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs

```diff
@@ -145,6 +148,211 @@ bool ExpectedErrors(WriteContext writeContext)
             }
         }
 
+        [Fact]
+        public async Task StreamIntsToServer()
+        {
+            using (StartVerifiableLog(out var loggerFactory, LogLevel.Trace))
+            {
+                var connection = new TestConnection();
+                var hubConnection = CreateHubConnection(connection, loggerFactory: loggerFactory);
+                await hubConnection.StartAsync().OrTimeout();
+
+                var channel = Channel.CreateUnbounded<int>();
+                var invokeTask = hubConnection.InvokeAsync<int>("SomeMethod", channel.Reader);
+
+                var invocation = await connection.ReadSentJsonAsync().OrTimeout();
+                Assert.Equal(HubProtocolConstants.InvocationMessageType, invocation["type"]);
+                Assert.Equal("SomeMethod", invocation["target"]);
+                var streamId = invocation["arguments"][0]["streamId"];
+
+                foreach (var number in new[] { 42, 43, 322, 3145, -1234 })
+                {
+                    await channel.Writer.WriteAsync(number);
```

Comment:
OrTimeout

Reply:
this one's a value task, so OrTimeout complains. Would just casting the task work?

Reply:
Hrm, we should add an `OrTimeout` for `ValueTask`. It can just do `.AsTask().OrTimeout()`

Reply:
Go ahead and add those if you're not already planning to. Feel free to grab me or @BrennanConroy if you need some guidance there. You should just be able to add them alongside the existing versions of `.OrTimeout()` and as I said, the implementation is pretty simple :)